lib/queue.rb

```diff
@@ -1,28 +1,38 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new
+    @front = 0 
+    @back = 0 
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == (@back + 1 ) % 20 
+    else  
+      @store.push(element)
+      @back += 1
+    end 
```

Comment:
It looks here like you _started_ to implement a circular buffer, but switched to implementing it with a straight array.  The assignment was to use a circular buffer.

```suggestion
    if @front == @back && @back == -1
      @front = @back = 0
    elsif @front == @back
      raise BufferFull.new "Queue is full"
    end
    
    @store[@back] = element
    @back = (@back +1) % @store.length
  end
```